e2e/playwright/point-click.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { test, expect } from './authenticatedAppFixture'
+
+// test file is for testing point an click code gen functionality that's not sketch mode related
+
+test('verify extruding circle works', async ({ app }) => {
+  const file = await app.getInputFile('test-circle-extrude.kcl')
+  await app.initialise(file)
+  const [clickCircle, moveToCircle] = app.makeMouseHelpers(582, 217)
+
+  await test.step('because there is sweepable geometry, verify extrude is enable when nothing is selected', async () => {
+    await app.clickNoWhere()
+    await app.expectExtrudeButtonToBeEnabled()
+  })
+
+  await test.step('check code model connection works and that button is still enable once circle is selected ', async () => {
+    await moveToCircle()
+    const circleSnippet =
+      'circle({ center: [318.33, 168.1], radius: 182.8 }, %)'
+    await app.expectCodeHighlightedToBe(circleSnippet)
+
+    await clickCircle()
+    await app.expectActiveLinesToBe([circleSnippet.slice(-5)])
+    await app.expectExtrudeButtonToBeEnabled()
+  })
+
+  await test.step('do extrude flow and check extrude code is added to editor', async () => {
+    await app.clickExtrudeButton()
+
+    await expect
+      .poll(() => app.serialiseCmdBar())
+      .toEqual({
+        currentArg: 'distance',
+        tabNames: ['Selection: 1 face', 'Distance:'],
+        currentTab: 'distance:',
+        currentArgValue: '5',
+        inReview: false,
+      })
+    await app.progressCmdBar()
+
+    const expectString = 'const extrude001 = extrude(5, sketch001)'
+    await app.expectEditor.not.toContain(expectString)
+
+    await expect
+      .poll(() => app.serialiseCmdBar())
+      .toEqual({
+        inReview: true,
+        tabNames: ['Selection: 1 face', 'Distance: 5'],
+        currentArg: '',
+        currentTab: '',
+        currentArgValue: '',
+      })
+    await app.progressCmdBar()
+
+    await app.expectEditor.toContain(expectString)
+  })
+})
```

Comment:
So hopefully this reads well, I'm not adhering 100% to some of Joel's suggestions, in part because that would require a lot more setup just to get going, (e.g. his whole app waiting concept)

But what I have done with this:
- No use of the playwright API at all in the test (expect doesn't count)
- No timeouts use in any part of this (including the fixture)
- Dabbled with the serialisation with the cmd bar

The serialisation of the cmd bar I find particularly interesting, because in the past I've basically just grabbed the bare minimum from the HTML to progress the test (check it says extrude somewhere or similar, chick the arrow button), but with the serialisation, it definitely help express what we expect the state of the cmd bar to be as it's progressed through.

Reply:
Talked a bit with Frank about improvements to the cmd bar serilsation, I may or may not make those changes in this PR.

the tabs for example should be called args and should be a key: value object.
We also want a union type for the three main states of the command bar, picking a command, adding arguments and reviewing.

Reply:
Okay I think it's better now
![image](https://github.com/user-attachments/assets/96e4f862-a04c-434b-a6a3-12f7d654f50a)